apps/teams-test-app/src/components/MenusAPIs.tsx

```diff
@@ -0,0 +1,84 @@
+import { menus } from '@microsoft/teams-js';
+import React from 'react';
+import { ReactElement } from 'react';
+import { ApiWithTextInput } from './utils';
+
+const SetUpViews = (): React.ReactElement =>
+  ApiWithTextInput<menus.ViewConfiguration[]>({
+    name: 'menusSetUpViews',
+    title: 'Set Up Views',
+    onClick: {
+      validateInput: input => {
+        input.forEach(viewConfig => {
+          if (!viewConfig?.id || !viewConfig.title) {
+            throw new Error('ID and Title are required for each viewConfiguration');
+          }
+        });
+      },
+      submit: async (input, setResult) => {
+        menus.setUpViews(input, id => {
+          setResult('handler called with id: ' + id);
+          return true;
+        });
+        return 'Completed';
```

Comment:
Is Menus a new API for v2 or does it also exist in v1? If it also exists in v1 we will need to add `withPromise` and `withCallback` calls to the `submit`.

Reply:
Discussed offline, this shouldn't be an issue. Resolving conversation